client/signup/steps/plans/index.jsx

```diff
@@ -30,12 +30,19 @@ export class PlansStep extends Component {
 	state = {
 		isDesktop: isDesktop(),
 		experiment: null,
+		experimentLoaded: false,
 	};
 
 	componentWillMount() {
```

Comment:
Hey there, folks! 

Do you think we could refactor this to not use `componentWillMount`? It's deprecated, and the next major version of React will stop working. Worst case scenario, we should update it to `UNSAFE_componentWillMount`, however those are also deprecated and we're doing a migration away from them in #58453. So ideally, we should use `componentDidMount` or if necessary, `static getDerivedStateFromProps()`. 

I'm at your disposal if you have any questions or need any help with this one. I was actually going to jump in and refactor it, but noticed that there has been pretty recent activity here and thought you folks might wanna address it.

Thanks in advance!

Reply:
@tyxla I've included your suggested change in https://github.com/Automattic/wp-calypso/pull/60182/. Could you please let me know if it looks good to you?

Reply:
Sure thing, I'll take a look 👀